query-service-impl/src/test/java/org/hypertrace/core/query/service/QueryRequestBuilderUtils.java

```diff
@@ -181,4 +181,70 @@ public static Filter createTimeFilter(String columnName, Operator op, long value
         .setOrder(order)
         .setExpression(createColumnExpression(columnName));
   }
+
+  public static Filter createFilter(
+      String columnName, Operator op, ValueType valueType, Object valueObject) {
+    ColumnIdentifier startTimeColumn =
+        ColumnIdentifier.newBuilder().setColumnName(columnName).build();
+    Expression lhs = Expression.newBuilder().setColumnIdentifier(startTimeColumn).build();
+
+    Value value = Value.newBuilder().setValueType(valueType).buildPartial();
+    switch (valueType) {
+      case LONG:
+        value = Value.newBuilder(value).setLong((long) valueObject).build();
+        break;
+      case INT:
+        value = Value.newBuilder(value).setInt((int) valueObject).build();
+        break;
+      case STRING:
+        value = Value.newBuilder(value).setString((String) valueObject).build();
+    }
+    LiteralConstant constant = LiteralConstant.newBuilder().setValue(value).build();
+    Expression rhs = Expression.newBuilder().setLiteral(constant).build();
+    return Filter.newBuilder().setLhs(lhs).setOperator(op).setRhs(rhs).build();
+  }
+
+  public static Function.Builder createTimeColumnGroupByFunction(
```

Comment:
Looks like it is copied twice - `createTimeColumnGroupByFunction`

Reply:
addressed in another comment.